mindone/visualize/videos.py

```diff
@@ -0,0 +1,37 @@
+import os
+
+import imageio
+import numpy as np
+
+
+def save_videos(frames: np.ndarray, path: str, fps=8, loop=1, concat=False):
+    """
+    Save video frames to gif files
+    Args:
+        frames: video frames in shape (b f h w 3), pixel value in [0, 1], RGB mode.
+        path:  file path to save the output gif
+        fps: frames per sencond in the output gif. 1/fps = display duration per frame
+        concat: if True and b>1, all videos will be concatnated in grids and saved as one gif.
+        loop: number of loops to play. If 0, it will play endlessly.
+    """
+    # input frames: (b f H W 3), normalized to [0, 1]
+    frames = (frames * 255).round().clip(0, 255).astype(np.uint8)
+    os.makedirs(os.path.dirname(path), exist_ok=True)
+
+    duration = 1 / fps
+    if len(frames.shape) == 4:
+        imageio.mimsave(path, frames, duration=duration)
```

Comment:
imageio's core v2 is obsolete and [will be deleted in the future](https://imageio.readthedocs.io/en/stable/reference/userapi.html#writing-images).
Use this instead:
```python
import imageio.v3 as iio
iio.imwrite(...)    # passed parameters are the same
```

Reply:
updated requirement